pkg/telemetry/cassandra_agent/cassandra_agent_config.go

```diff
@@ -71,41 +70,19 @@ func (c Configurator) ReconcileTelemetryAgentConfig(dc *cassdcapi.CassandraDatac
 	if err != nil {
 		return result.Error(err)
 	}
-	cmObjectKey := types.NamespacedName{
-		Name:      c.Kluster.Name + "-" + c.DcName + "-metrics-agent-config",
-		Namespace: c.DcNamespace,
-	}
 	KlKey := types.NamespacedName{
 		Name:      c.Kluster.Name,
 		Namespace: c.Kluster.Namespace,
 	}
 	partOfLabels := labels.PartOfLabels(KlKey)
 	desiredCm.SetLabels(partOfLabels)
-	annotations.AddHashAnnotation(desiredCm)
-
-	currentCm := &corev1.ConfigMap{}
-
-	err = c.RemoteClient.Get(c.Ctx, cmObjectKey, currentCm)
-
-	if err != nil {
-		if errors.IsNotFound(err) {
-			if err := c.RemoteClient.Create(c.Ctx, desiredCm); err != nil {
-				return result.Error(err)
-			}
-			return result.RequeueSoon(c.RequeueDelay)
-		} else {
-			return result.Error(err)
-		}
-	}
 
-	if !annotations.CompareHashAnnotations(currentCm, desiredCm) {
-		resourceVersion := currentCm.GetResourceVersion()
-		desiredCm.DeepCopyInto(currentCm)
-		currentCm.SetResourceVersion(resourceVersion)
-		if err := c.RemoteClient.Update(c.Ctx, currentCm); err != nil {
-			return result.Error(err)
-		}
-		return result.RequeueSoon(c.RequeueDelay)
+	recRes := reconciliation.ReconcileConfigMap(c.Ctx, c.RemoteClient, c.RequeueDelay, *desiredCm)
+	switch {
+	case recRes.IsError():
+		return recRes
+	case recRes.IsRequeue():
+		return recRes
```

Comment:
nit: Why us a switch instead of an `if` ?

```suggestion
	if recRes.IsError() || recRes.IsRequeue(){
		return recRes
	}
```

Reply:
Firstly, because I think it is easier to read than an if/else. Secondly, because longer term I think we should think about making the reconcile.Result types public so that we don't have to have so many silly isDone/isRequeue type helper functions, as they aren't that maintainable.

Reply:
Oh, and also, the switch is needed in other places where you (for e.g.) don't return an error. I was just keeping it consistent in this case where you directly return the result.